server/task_scheduler.go

```diff
@@ -292,6 +292,45 @@ func (s *TaskScheduler) Register(taskType api.TaskType, executorGetter func() Ta
 	s.executorGetters[taskType] = executorGetter
 }
 
+// ClearRunningTasks changes all RUNNING tasks to CANCELED.
+// When there are running tasks and Bytebase server is shutdown, these task executors are stopped, but the tasks' status are still RUNNING.
+// When Bytebase is restarted, the task scheduler will re-schedule those RUNNING tasks, which should be CANCELED instead.
+// So we change their status to CANCELED before starting the scheduler.
+func (s *TaskScheduler) ClearRunningTasks(ctx context.Context) error {
+	taskFind := &api.TaskFind{StatusList: &[]api.TaskStatus{api.TaskRunning}}
+	runningTasks, err := s.server.store.FindTask(ctx, taskFind, false)
+	if err != nil {
+		return errors.Wrap(err, "failed to get running tasks")
+	}
+	for _, task := range runningTasks {
+		if _, err := s.server.store.PatchTaskStatus(ctx, &api.TaskStatusPatch{
+			ID:        task.ID,
+			UpdaterID: api.SystemBotID,
+			Status:    api.TaskCanceled,
```

Comment:
I had a hard time thinking whether the status should be cancelled or failed. Some things can be left in incomplete or inconsistent state. Let's do cancelled for now but probably need to provide documentation for users handle such situations.